### PR TITLE
EZR-3984: refactored touched classes and added support for create/get…

### PR DESCRIPTION
**Pull request by fatebinder**
… attachments, templates, automated communications

### CODE REVIEW

File: src/Request/CreateAttachment.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    protected $url = '/filemanager/files/%s';
+    /**
+     * @var string
+     */
+    public $method = 'POST';
+    /**
+     * @var bool
+     */
+    public $requiresAuth = true;
```
Comment (mortenhauberg) [inline src/Request/CreateAttachment.php]: `true` by default

File: src/Request/CreateAttachment.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * @var string
+     */
+    public $responseClass = Attachment::class;
+    /**
+     * @var string
+     */
+    protected $modelClass = CreateAttachmentModel::class;
+
+    public $options = ['header' => ['Content-Type' => 'multipart/form-data']];
```
Comment (mortenhauberg) [inline src/Request/CreateAttachment.php]: Missing doc block

File: src/Request/CreateAttachment.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    public function __construct(RequestModel $model = null)
```
Comment (mortenhauberg) [inline src/Request/CreateAttachment.php]: Missing doc block

File: src/Request/CreateAutomatedCommunication.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+     */
+    protected $url = '/automated-communications';
+    /**
+     * @var string
+     */
+    public $method = 'POST';
+    /**
+     * @var bool
+     */
+    public $requiresAuth = true;
```
Comment (mortenhauberg) [inline src/Request/CreateAutomatedCommunication.php]: `true` by default

File: src/Request/CreateTemplate.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+     */
+    protected $url = '/templates';
+    /**
+     * @var string
+     */
+    public $method = 'POST';
+    /**
+     * @var bool
+     */
+    public $requiresAuth = true;
```
Comment (mortenhauberg) [inline src/Request/CreateTemplate.php]: `true` by default

File: src/Request/GetAttachment.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    protected $url = '/filemanager/files/%s';
+    /**
+     * @var string
+     */
+    public $method = 'GET';
+    /**
+     * @var bool
+     */
+    public $requiresAuth = true;
```
Comment (mortenhauberg) [inline src/Request/GetAttachment.php]: `true` by default

File: src/Request/GetAutomatedCommunication.php [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    protected $url = '/automated-communications/%s';
+    /**
+     * @var string
+     */
+    public $method = 'GET';
+    /**
+     * @var bool
+     */
+    public $requiresAuth = true;
```
Comment (mortenhauberg) [inline src/Request/GetAutomatedCommunication.php]: `true` by default

File: src/Request/GetRelationPath.php [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    protected $url = '/relations-paths/%s';
+    /**
+     * @var string
+     */
+    public $method = 'GET';
+    /**
+     * @var bool
+     */
+    public $requiresAuth = true;
```
Comment (mortenhauberg) [inline src/Request/GetRelationPath.php]: `true` by default

File: src/Request/GetTemplate.php [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    protected $url = '/templates/%d';
+    /**
+     * @var string
+     */
+    public $method = 'GET';
+    /**
+     * @var bool
+     */
+    public $requiresAuth = true;
```
Comment (mortenhauberg) [inline src/Request/GetTemplate.php]: `true` by default

File: src/Request/Model/CreateCompany.php [hunk 10]
```diff
@@ -4,14 +4,70 @@
...
-}
\ No newline at end of file
+    /**
+     * @var string
+     */
+    public $name;
+    /**
+     * @var string
+     */
+    public $timezone = "Europe/Copenhagen";
```
Comment (mortenhauberg) [inline src/Request/Model/CreateCompany.php]: Use single quotes :)

Comment (mortenhauberg) [inline src/Request/Model/CreateCompany.php]: Should we have defaults for timezone, currency etc..?

File: src/Model/Field.php [hunk 11]
```diff
@@ -9,12 +9,11 @@
     public $name, $value;
 
     /**
-     * @static
      * @param string $name
      * @param string $value
      * @return static
```
Comment (mortenhauberg) [inline src/Model/Field.php]: `return self`

File: src/Model/Field.php [hunk 12]
```diff
@@ -9,12 +9,11 @@
...
      */
     public static function create($name, $value) {
         return new static(["name" => $name, "value" => $value]);
```
Comment (mortenhauberg) [inline src/Model/Field.php]: Single quotes

File: src/Model/LeadData.php [hunk 13]
```diff
@@ -6,9 +6,21 @@
...
+    public $key;
+    /**
+     * @var string
+     */
+    public $value;
 
+    /**
+     * @param $key
+     * @param $value
+     * @return static
```
Comment (mortenhauberg) [inline src/Model/LeadData.php]: `return self`

File: src/Model/LeadData.php [hunk 14]
```diff
@@ -6,9 +6,21 @@
...
+     * @var string
+     */
+    public $key;
+    /**
+     * @var string
+     */
+    public $value;
 
+    /**
+     * @param $key
```
Comment (mortenhauberg) [inline src/Model/LeadData.php]: Types

File: src/Model/LeadData.php [hunk 15]
```diff
@@ -6,9 +6,21 @@
...
+     * @param $value
+     * @return static
+     */
     public static function create($key, $value) {
         return new static(["key" => $key, "value" => $value]);
```
Comment (mortenhauberg) [inline src/Model/LeadData.php]: Single quotes

File: src/Model/LeadType.php [hunk 16]
```diff
@@ -6,8 +6,68 @@
...
+    /**
+     * @var string
+     */
+    public $id;
+    /**
+     * @var string
+     */
+    public $name;
+    /**
+     * @var string //is used?
```
Comment (mortenhauberg) [inline src/Model/LeadType.php]: Well.. Is it?

File: src/Model/LeadType.php [hunk 17]
```diff
@@ -6,8 +6,68 @@
...
+    /**
+     * @var int
+     */
+    public $escalation_activity;
+    /**
+     * @var bool
+     */
+    public $disabled;
+    /**
+     * @var bool //is used?
```
Comment (mortenhauberg) [inline src/Model/LeadType.php]: ??

File: src/Model/LeadType.php [hunk 18]
```diff
@@ -6,8 +6,68 @@
...
+     */
+    public $highlighted;
+    /**
+     * how does it look?
```
Comment (mortenhauberg) [inline src/Model/LeadType.php]: Invalid doc block

File: src/Model/Option.php [hunk 19]
```diff
@@ -6,15 +6,21 @@
...
+    /**
+     * @var string
+     */
+    public $value;
 
     /**
-     * @static
      * @param string $id
      * @param string $value
      * @return static
```
Comment (mortenhauberg) [inline src/Model/Option.php]: `return self`

File: src/Model/Rights.php [hunk 20]
```diff
@@ -6,6 +6,16 @@
 
 class Rights extends Model
 {
-    public $dashboard, $contacts, $opportunities, $cases, $tasks, $reports, $campaigns,
-            $products, $agency, $settings, $emails, $notifications;
-}
\ No newline at end of file
+    public $dashboard;
```
Comment (mortenhauberg) [inline src/Model/Rights.php]: Missing doc blocks

File: src/Model/Role.php [hunk 21]
```diff
@@ -6,5 +6,25 @@
...
\ No newline at end of file
+    /**
+     * @var int
+     */
+    public $id;
+    /**
+     * @var string
+     */
+    public $name;
+    public $access_teams;
```
Comment (mortenhauberg) [inline src/Model/Role.php]: Missing doc block

File: src/Model/Team.php [hunk 22]
```diff
@@ -6,6 +6,77 @@
...
+    public $vat;
+    /**
+     * @var Timezone
+     */
+    public $timezone;
+    /**
+     * @var Locale
+     */
+    public $locale;
+    public $group;
```
Comment (mortenhauberg) [inline src/Model/Team.php]: Missing doc block

File: src/Model/TeamSettings.php [hunk 23]
```diff
@@ -6,7 +6,14 @@
...
 {
+    /**
+     * @var array
+     */
     protected $collections = ['escalation_active' => 'escalation_active'];
 
-    public $users, $disabled, $escalation_active, $escalation_accept, $escalation_activity;
-}
\ No newline at end of file
+    public $users;
```
Comment (mortenhauberg) [inline src/Model/TeamSettings.php]: Missing doc blocks

File: src/Model/User.php [hunk 24]
```diff
@@ -6,7 +6,70 @@
 
 class User extends Response
 {
-    public $id, $first_name, $last_name, $phone_number, $job_title, $email, $email_notifications,
-            $profile_picture, $user_type, $locale, $timezone, $role, $company, $settings,
-            $team, $signature, $created_at, $updated_at;
-}
\ No newline at end of file
+    protected $collections = ['roles' => 'role'];
```
Comment (mortenhauberg) [inline src/Model/User.php]: Missing doc block

File: src/Model/User.php [hunk 25]
```diff
@@ -6,7 +6,70 @@
...
+     */
+    public $system_user;
+    /**
+     * @var string
+     */
+    public $created_at;
+    /**
+     * @var string
+     */
+    public $updated_at;
```
Comment (mortenhauberg) [inline src/Model/User.php]: Missing doc blocks

File: src/Request/CreateAttachment.php [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * @var string
+     */
+    public $responseClass = Attachment::class;
+    /**
+     * @var string
+     */
+    protected $modelClass = CreateAttachmentModel::class;
+    /**
+     * @var mixed
```
Comment (mortenhauberg) [inline src/Request/CreateAttachment.php]: Doesn't look mixed

File: src/Request/CreateAttachment.php [hunk 27]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    public $options = ['header' => ['Content-Type' => 'multipart/form-data']];
+
+    /**
+     * CreateAttachment constructor.
```
Comment (mortenhauberg) [inline src/Request/CreateAttachment.php]: I don't think this comment is needed

File: src/Request/Model/CreateCompany.php [hunk 28]
```diff
@@ -4,14 +4,70 @@
...
-}
\ No newline at end of file
+    /**
+     * @var string
+     */
+    public $name;
+    /**
+     * @var string
+     */
+    public $timezone = 'Europe/Copenhagen';
```
Comment (mortenhauberg) [inline src/Request/Model/CreateCompany.php]: Do we need defaults?

File: src/Request/Model/CreateTeam.php [hunk 29]
```diff
@@ -13,7 +13,7 @@ class CreateTeam extends Model
     /**
      * @var string
      */
-    public $timezone = "Europe/Copenhagen";
+    public $timezone = 'Europe/Copenhagen';
```
Comment (mortenhauberg) [inline src/Request/Model/CreateTeam.php]: Do we need defaults?

File: src/Request/Model/CreateUser.php [hunk 30]
```diff
@@ -33,7 +33,7 @@ class CreateUser extends Model
     /**
      * @var string
      */
-    public $timezone = "Europe/Copenhagen";
+    public $timezone = 'Europe/Copenhagen';
```
Comment (mortenhauberg) [inline src/Request/Model/CreateUser.php]: Defaults?

File: src/Model/LeadType.php (hunk 16 above)
Comment (fatebinder) [inline src/Model/LeadType.php]: I haven't seen it in the docs, but I haven't removed it from here cuz I don't know where its used.

Comment (fatebinder) [inline src/Model/LeadType.php]: It is used, in the response :)

File: src/Model/Role.php (hunk 21 above)
Comment (fatebinder) [inline src/Model/Role.php]: haven't found it in docs

File: src/Model/Team.php (hunk 22 above)
Comment (fatebinder) [inline src/Model/Team.php]: haven't found it in the response from zebra or on docs. Maybe its a typo from group_id?

File: src/Model/TeamSettings.php (hunk 23 above)
Comment (fatebinder) [inline src/Model/TeamSettings.php]: Don't know where to find them in docs ...

Comment (mortenhauberg) [inline src/Model/TeamSettings.php]: Look in the code. Look in the actual response. That goes for everything you can't find in the docs. If you think it should be in the docs, then create a ticket for it

File: src/Model/LeadData.php [hunk 31]
```diff
@@ -6,9 +6,21 @@
...
+     * @var string
+     */
+    public $key;
+    /**
+     * @var string
+     */
+    public $value;
 
+    /**
+     * @param string
```
Comment (mortenhauberg) [inline src/Model/LeadData.php]: Missing variable names

File: src/Model/LocalizationTypes.php [hunk 32]
```diff
@@ -18,6 +16,26 @@ class LocalizationTypes extends Response
...
+    public $languages;
+    /**
+     * @var Country[]
+     */
+    public $countries;
 
-    public $locales, $timezones, $currencies, $languages, $countries, $plan;
-}
\ No newline at end of file
+    public $plan;
```
Comment (mortenhauberg) [inline src/Model/LocalizationTypes.php]: Missing doc block

File: src/Model/Team.php (hunk 22 above)
Comment (mortenhauberg) [inline src/Model/Team.php]: ping

File: src/Model/User.php (hunk 24 above)
Comment (mortenhauberg) [inline src/Model/User.php]: ping

File: src/Model/User.php (hunk 25 above)
Comment (mortenhauberg) [inline src/Model/User.php]: ping

**Pull request merged**